backend/dataall/modules/dataset_sharing/services/share_managers/s3_access_point_share_manager.py

```diff
@@ -146,97 +146,71 @@ def grant_target_role_access_policy(self):
         logger.info(
             f'Grant target role {self.target_requester_IAMRoleName} access policy'
         )
+
+        share_resource_policy_name = ConsumptionRole.generate_policy_name(self.target_environment.environmentUri,
+                                                                          self.target_requester_IAMRoleName)
+        version_id, policy_document = IAM.get_managed_policy_default_version(
+            self.target_account_id,
+            share_resource_policy_name)
+
         key_alias = f"alias/{self.dataset.KmsAlias}"
         kms_client = KmsClient(self.dataset_account_id, self.source_environment.region)
         kms_key_id = kms_client.get_key_id(key_alias)
 
-        existing_policy = IAM.get_role_policy(
-            self.target_account_id,
-            self.target_requester_IAMRoleName,
-            IAM_ACCESS_POINT_ROLE_POLICY,
+        s3_target_resources = [
+            f"arn:aws:s3:::{self.bucket_name}",
+            f"arn:aws:s3:::{self.bucket_name}/*",
+            f"arn:aws:s3:{self.dataset_region}:{self.dataset_account_id}:accesspoint/{self.access_point_name}",
+            f"arn:aws:s3:{self.dataset_region}:{self.dataset_account_id}:accesspoint/{self.access_point_name}/*"
+        ]
+
+        share_manager = ShareManagerUtils(
+            self.session,
+            self.dataset,
+            self.share,
+            self.source_environment,
+            self.target_environment,
+            self.source_env_group,
+            self.env_group
+        )
+        share_manager.add_missing_resources_to_policy_statement(
+            self.bucket_name,
+            s3_target_resources,
+            policy_document["Statement"][0],
+            IAM_ACCESS_POINT_ROLE_POLICY
```

Comment:
Passing `IAM_ACCESS_POINT_ROLE_POLICY` here is only for logging statements but is to be `iam_role_policy_name` which is no longer the case with this PR - I think we should update what we pass to `add_missing_resources_to_policy_statement(...)` and may no longer need `IAM_ACCESS_POINT_ROLE_POLICY`

Reply:
changed